internal/services/workloads/workloads_sap_virtual_instance_resource.go

```diff
@@ -0,0 +1,1129 @@
+														},
+
+														"load_balancer": SchemaForSAPVirtualInstanceLoadBalancerFullResourceNames(),
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"database_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validate.AvailabilitySetName,
+														},
+
+														"load_balancer": SchemaForSAPVirtualInstanceLoadBalancerFullResourceNames(),
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"shared_storage": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"account_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: storageValidate.StorageAccountName,
+														},
+
+														"private_endpoint_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: networkValidate.PrivateLinkName,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+
+								"high_availability_type": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringInSlice(sapvirtualinstances.PossibleValuesForSAPHighAvailabilityType(), false),
+								},
+
+								"is_secondary_ip_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Optional: true,
+									ForceNew: true,
+									Default:  false,
+								},
+
+								"transport_create_and_mount": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									ForceNew: true,
+									MaxItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"resource_group_name": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ForceNew:     true,
+												ValidateFunc: resourcegroups.ValidateName,
+											},
+
+											"storage_account_name": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ForceNew:     true,
+												ValidateFunc: storageValidate.StorageAccountName,
```

Comment:
Can these also be IDs?

Reply:
"storage_account_name" means service API will create new storage account per name. So I assume we can't convert it to xxx_id.

I updated "resource_group_name" to "resource_group_id" since service API requires that it exists.